### PR TITLE
Feat/threads

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

* Implements the possibility for the service to publish threads
* Fixes an issue with round transitions
* Fixes a configuration issue

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: packages/valory/connections/twitter/connection.py [hunk 1]
```diff
@@ -180,7 +180,19 @@ def create_tweet(
...
+                for tweet in text:
+                    if not previous_tweet_id:
+                        response = api.create_tweet(text=tweet)
+                    else:
+                        response = api.create_tweet(text=tweet, in_reply_to_tweet_id=previous_tweet_id)
+                    previous_tweet_id = response.data["id"]
+            else:
+                # Single tweet
+                response = api.create_tweet(text=text)
+
```
Comment (dvilelaf) [inline packages/valory/connections/twitter/connection.py:195]: Thread implementation: if the data passed is a list, we post it as a thread.

File: packages/valory/services/impact_evaluator/service.yaml [hunk 2]
```diff
@@ -80,7 +80,7 @@ extra:
         termination_sleep: ${TERMINATION_SLEEP:int:900}
         reset_pause_duration: ${RESET_PAUSE_DURATION:int:300}
         on_chain_service_id: ${ON_CHAIN_SERVICE_ID:int:null}
-        reset_tendermint_after: ${RESET_TENDERMINT_AFTER:int:300}
+        reset_tendermint_after: ${RESET_TENDERMINT_AFTER:int:30}
```
Comment (dvilelaf) [inline packages/valory/services/impact_evaluator/service.yaml:83]: Small config fix

File: packages/valory/skills/impact_evaluator_abci/fsm_specification.yaml [hunk 3]
```diff
@@ -145,13 +145,13 @@ transition_func:
...
-    (TwitterHashtagsCollectionRound, ROUND_TIMEOUT): TwitterHashtagsCollectionRound
+    (TwitterHashtagsCollectionRound, NO_MAJORITY): TwitterRandomnessRound
+    (TwitterHashtagsCollectionRound, ROUND_TIMEOUT): TwitterRandomnessRound
     (TwitterMentionsCollectionRound, API_ERROR): TwitterMentionsCollectionRound
     (TwitterMentionsCollectionRound, DONE): TwitterDecisionMakingRound
     (TwitterMentionsCollectionRound, DONE_MAX_RETRIES): TwitterDecisionMakingRound
-    (TwitterMentionsCollectionRound, NO_MAJORITY): TwitterMentionsCollectionRound
-    (TwitterMentionsCollectionRound, ROUND_TIMEOUT): TwitterMentionsCollectionRound
+    (TwitterMentionsCollectionRound, NO_MAJORITY): TwitterRandomnessRound
+    (TwitterMentionsCollectionRound, ROUND_TIMEOUT): TwitterRandomnessRound
```
Comment (dvilelaf) [inline packages/valory/skills/impact_evaluator_abci/fsm_specification.yaml:154]: We should select new keepers whenever we time out. I found an issue today where the service was stuck in a timeout loop.

File: packages/valory/skills/twitter_scoring_abci/rounds.py [hunk 4]
```diff
@@ -607,15 +607,15 @@ class TwitterScoringAbciApp(AbciApp[Event]):
...
+            Event.ROUND_TIMEOUT: TwitterRandomnessRound,
         },
         TwitterHashtagsCollectionRound: {
             Event.DONE: TwitterDecisionMakingRound,
             Event.DONE_MAX_RETRIES: TwitterDecisionMakingRound,
             Event.API_ERROR: TwitterHashtagsCollectionRound,
-            Event.NO_MAJORITY: TwitterHashtagsCollectionRound,
-            Event.ROUND_TIMEOUT: TwitterHashtagsCollectionRound,
+            Event.NO_MAJORITY: TwitterRandomnessRound,
+            Event.ROUND_TIMEOUT: TwitterRandomnessRound,
```
Comment (dvilelaf) [inline packages/valory/skills/twitter_scoring_abci/rounds.py:618]: We should select new keepers whenever we time out. I found an issue today where the service was stuck in a timeout loop.

**Review by 0xArdi — APPROVED** [review]

**Review by angrybayblade — APPROVED** [review]

**Pull request merged**